src/base64.cpp

```diff
@@ -0,0 +1,52 @@
+#include <migraphx/base64.hpp>
+#include <vector>
+#include <array>
+#include <iostream>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+namespace {
+
+std::array<char, 64> constexpr B64chars{
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+    'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+    'w', 'x', 'y', 'z', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '/'};
+
+/// base64 encoder snippet altered from https://stackoverflow.com/a/37109258
+const std::string b64_encode(const void* data, const size_t& len)
+{
+    std::vector<char> res_vec((len + 2) / 3 * 4, '=');
+    const unsigned char* p = static_cast<const unsigned char*>(data);
+    size_t j = 0, pad = len % 3;
+    const size_t last = len - pad;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunsafe-buffer-usage"
+    for(size_t i = 0; i < last; i += 3)
+    {
+        unsigned n =
+            static_cast<unsigned>(p[i]) << 16 | static_cast<unsigned>(p[i + 1]) << 8 | p[i + 2];
```

Comment:
What happens if the input is of len `1`. This will read at `p[0]`, but trip beyond: at `p[1]` and `p[2]`?

Reply:
If len =1, pad = 1, then last = 0. So it skips the for loop. Same for if len = 2, pad = 2, last = 0.